breakwater/src/main.rs

```diff
@@ -106,19 +69,20 @@ async fn main() -> Result<(), Error> {
         args.network_buffer_size
             .try_into()
             // This should never happen as clap checks the range for us
-            .context(InvalidNetworkBufferSizeSnafu {
-                network_buffer_size: args.network_buffer_size,
-            })?,
+            .context(format!(
```

Comment:
nit: AFAIK when doing actual string allocations/formatting one should prefer `with_context` over `context`.
This way the string formatting is only run lazily in the error case.

This applies to all format!() invocations below. In this specific line the performance is unimportant, I personally try to stick to it regardless, just for good habits.